prediction_market_agent/agents/microchain_agent/blockchain/contract_class_converter.py

```diff
@@ -50,9 +50,13 @@ def create_class(
 class ContractClassConverter:
     """Class responsible for reading a smart contract on Gnosis Chain and converting its functionalities into Python classes."""
 
-    def __init__(self, contract_address: ChecksumAddress):
+    contract_address: ChecksumAddress
```

Comment:
I see why do this if you are using the `self.build_x()` pattern in __init__, but if you do in the __init__ just 

```
 self.contract_address = contract_address
self.contract_name = contract_name
```

there is no need for this.

```suggestion
```

Reply:
Added Pydantic Base Model, hope it fixes this?

Reply:
The pydantic base model is meant to hold some data; it's like a typed dictionary on steroids. This seems to be a regular class that encapsulates some functions, so pydantic shouldn't be used here.

If these two lines are removed, all is okay.

Reply:
done